vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -95,6 +111,107 @@ def __preprocess(
+                    arg_dict = json.loads(tool_call_function["arguments"])
+                    new_tool_call_function = GraniteGuardianToolCallFunctionObject(
+                        name=tool_call_function["name"], arguments=arg_dict
+                    )
+                    assistant_tool_call_functions.append(new_tool_call_function)
+                assistant_content_string = json.dumps(
+                    assistant_tool_call_functions, indent=self.INDENT
+                )
+                assistant_message = DetectionChatMessageParam(
+                    role=message["role"], content=assistant_content_string
+                )
+                assistant_message_idxs.append(i)
+
+            if message["role"] == "user":
+                user_message = message
+                user_message_idxs.append(i)
+
+        # Error if no assistant message was found
+        if not assistant_message or not assistant_message["content"]:
+            return ErrorResponse(
+                message="no assistant message was provided with tool_calls for analysis",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+        # Error if no user message was found
+        if not user_message or not user_message["content"]:
+            return ErrorResponse(
+                message="no user message was provided with content for analysis",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+        # Warning if multiple assistant messages were found
+        if len(assistant_message_idxs) > 1:
+            logger.warning(
+                "More than one assistant message with tool_calls was provided. Only the last will be used for analysis with tools."
+            )
+        # Warning if multiple user messages were found
+        if len(user_message_idxs) > 1:
+            logger.warning(
+                "More than one user message was provided. Only the last will be used for analysis with tools."
+            )
+
+        # Provide the inner tools functions as a message with role: tools
+        tools_functions = []
+        for tools in request.tools:
+            tools_functions.append(tools["function"])
+        tools_content_string = json.dumps(tools_functions, indent=self.INDENT)
+        tools_message = DetectionChatMessageParam(
+            role="tools", content=tools_content_string
+        )
+
+        # Replace request portions
+        # `tools` should not be passed onto the chat completions request currently, but we
+        # do not pass them on here anyway, in case the chat completion request processing
+        # behavior changes, this could affect the completion generation.
+        request.tools = []
+        # `tools` and `user` messages are more interchangeable order-wise but the assistant
+        # message needs to be last.
+        request.messages = [tools_message, user_message, assistant_message]
```

Comment:
does the changed ordering of these messages, matter? like the tool may be in 2nd message and assistant message in 1st message in the input request to the detector adapter ?